### PR TITLE
Allow passing a ref (innerRef) to a Checkbox

### PR DESCRIPTION
**Pull request by tusbar**
I believe it is time to think about implementing ref forwarding – so that passing a ref always does what you expect.

I have started using https://github.com/react-hook-form/react-hook-form – and it would be nice if this library would support it as the API is really great.

### CODE REVIEW

File: src/checkbox/src/Checkbox.js [hunk 1]
```diff
@@ -104,13 +109,15 @@ class Checkbox extends PureComponent {
...
+    innerRef: () => {},
     onChange: () => {},
     appearance: 'default'
   }
 
-  setIndeterminate = el => {
+  handleInnerRef = el => {
     if (!el) return
     el.indeterminate = this.props.indeterminate
+    this.props.innerRef(el)
```
Comment (mshwery) [inline src/checkbox/src/Checkbox.js]: wouldn't you want to bubble the `innerRef` even when `!el`?

```suggestion
    if (el) {
      el.indeterminate = this.props.indeterminate
    }
    this.props.innerRef(el)
```

**Comment by mshwery** [comment]
Also I agree that we should start adding support for ref forwarding.

File: src/checkbox/src/Checkbox.js (hunk 1 above)
Comment (tusbar) [reply src/checkbox/src/Checkbox.js]: Yeah, that makes sense.

**Comment by tusbar** [comment]
Fixed.

File: src/checkbox/src/Checkbox.js [hunk 2]
```diff
@@ -73,6 +73,11 @@ class Checkbox extends PureComponent {
      */
     indeterminate: PropTypes.bool,
 
+    /**
+     * Function that returns the ref of the checkbox.
+     */
+    innerRef: PropTypes.func,
```
Comment (mshwery) [inline src/checkbox/src/Checkbox.js]: Can you update the `index.d.ts` typedef too?

Comment (tusbar) [reply src/checkbox/src/Checkbox.js]: 👍

**Pull request merged**